src/web/crate_details.rs

```diff
@@ -348,7 +348,9 @@ mod tests {
 
     #[test]
     fn test_last_successful_build_when_last_release_failed() {
-        crate::test::with_database(|db| {
+        crate::test::wrapper(|env| {
```

Comment:
It might be easier to add a `crate::test::with_db` function that does exactly this: `crate::test::wrapper(|env| f(env.db()))`. That would make the tests easier to write.

Reply:
Hmm, that was the approach I went with before that commit, and I wasn't much satisfied with it: unless we want to special-case db tests (why?) we're going to eventually have a bunch of combinations (`db`, `frontend`, an eventual `build`...) or nested wrappers, both of which I don't really like.

I don't think this is too bad: if you don't want to add the `let db = env.db()` you can always use `env.db().conn()...`.

Reply:
I agree a combinatorial explosion would be bad. I was imagining a linear number of methods: `db()`, `frontend()`, `build()`,... and if you want to do anything fancier you call `test::wrapper` yourself. 

It came to mind because I saw you were changing this a lot below, but it's not a giant deal.